src/Development/Shake/Internal/Derived.hs

```diff
@@ -72,22 +75,30 @@ addShakeExtra x = Map.insert (typeOf x) (toDyn x)
 
 -- | @copyFile' old new@ copies the existing file from @old@ to @new@.
 --   The @old@ file will be tracked as a dependency.
+--   Also creates the new directory if necessary.
 copyFile' :: FilePath -> FilePath -> Action ()
 copyFile' old new = do
     need [old]
     putLoud $ "Copying from " ++ old ++ " to " ++ new
+    createDirIfMissing new
```

Comment:
@kirbyfan64 - note that createDirIfMissing is a helper that creates the directory of the file, rather than `new` itself. Given that in the other commit I suggest removing most of the logic of `createDirIfMissing`, I'd recommend inlining whatever is left, as it will make it more clear what it is doing.